jetty-server/src/main/java/org/eclipse/jetty/server/handler/ResourceHandler.java

```diff
@@ -86,9 +86,18 @@ public String getWelcomeFile(String pathInContext)
         for (int i = 0; i < _welcomes.length; i++)
         {
             String welcomeInContext = URIUtil.addPaths(pathInContext, _welcomes[i]);
-            Resource welcome = getResource(welcomeInContext);
-            if (welcome != null && welcome.exists())
-                return welcomeInContext;
+            try
+            {
+                Resource welcome = getResource(welcomeInContext);
+                if (welcome.exists())
+                    return welcomeInContext;
+            }
+            catch (IOException e)
```

Comment:
again let's analyse what happens without a catch here

Reply:
In the past, if `welcomeInContext` resulted in an invalid resource, it would return null. (logging the various exceptions in the `.getResource()` call.

Now, if the Resource can be created, it will be created and then tested if it exists.
Otherwise if the Resource cannot be created (bad URL syntax, invalid characters on the filesystem, etc) it throws.

Reply:
So in what circumstances would `getResource(URIUtil.addPaths(pathInContext, _welcomes[i])` throw?
Only if the pathInContext or the welcome file we badly formatted, which I think is a configuration error and should not be caught and ignored.      I think it is fine to not catch here..... unless you can think of a valid configuration of welcome files that would sometimes throw?